vm/systemSmartContracts/eei.go

```diff
@@ -144,13 +144,17 @@ func (host *vmContext) GetStorageFromAddress(address []byte, key []byte) []byte
 		if value, isInMap := storageAdrMap[string(key)]; isInMap {
 			return value
 		}
+	} else {
```

Comment:
I'm not sure I understand the reason behind these changes. They bring no utility from my point of view.
If you have **no storage** => you create empty map and fill it here : `storageAdrMap[string(key)] = data`, but this code has no effect on the internal data, nor the returned result. Why was this necessary?

Reply:
this is an optimization. I just saw that it could be added here - it can be reverted.

Reply:
You need to create the internal map, otherwise it is nil. and yields in error.